lib/tasks/production.rake

```diff
@@ -155,25 +155,32 @@ namespace :production do
     version = release_version(args[:version])
     remote  = args[:remote] || "upstream"
 
+    puts "Deploying version #{version}..."
+
+    puts
     puts "Ensuring version number..."
+    lines = File.readlines("templates/bot.yaml")
+    lines.map! { |line| line.gsub(/(image: .+miq_bot:)v\d+\.\d+\.\d/, "\\1#{version}") }
+    File.write("templates/bot.yaml", lines.join)
+    unless system("git diff --quiet templates/bot.yaml")
+      exit 1 unless system("git add templates/bot.yaml && git commit -m 'Release #{version}'")
+    end
     unless File.readlines("templates/bot.yaml").grep(/image: .+miq_bot:v/).all? { |l| l.include?("miq_bot:#{version}") }
       $stderr.puts "ERROR: images in templates/bot.yaml have not been updated to the expected version"
       exit 1
     end
 
-    puts "Deploying version #{version}..."
-
     puts
     puts "Tagging repository..."
     if `git tag --list #{version}`.chomp.empty?
-      exit 1 unless system("git tag #{version}")
+      exit 1 unless system("git tag #{version} -m 'Release #{version}'")
```

Comment:
The main reason is that I do signed tags, and signed tags require annotation, which requires a commit message.  If someone running this doesn't have that signing locally it will work the same, but just won't be signed. However, in the reverse, if I don't have this commit change and tag signing is set up, then the person running it gets a vi pop-up while it's running.

Reply:
I could also change this code to enforce not signing on this particular tag, but I think supporting signed tags is a better way forward.